src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1660,7 +1668,11 @@ private void EmitStaticCallExpression(BoundCall call, UseKind useKind)
 
             Debug.Assert(method.IsStatic);
 
-            EmitArguments(arguments, method.Parameters, call.ArgumentRefKindsOpt);
+            EmitArguments(
+                arguments,
+                method.Parameters,
+                call.ArgumentRefKindsOpt,
+                mightEscapeTemporaryRefs: MightEscapeTemporaryRefs(call, used: useKind != UseKind.Unused, receiverAddressKind: null));
```

Comment:
>mightEscapeTemporaryRefs: MightEscapeTemporaryRefs(call, used: useKind != UseKind.Unused, receiverAddressKind: null) [](http://example.com/codeflow?start=16&length=116)

Instead of threading an extra parameter through many layers, consider simply truncating ```_expressionTemps``` after the ```EmitArguments``` call as ```EmitAssignmentValue``` does.